### PR TITLE
Promote `UseDNSRecords` feature gate to GA

### PR DESCRIPTION
**Pull request by acumino**
**How to categorize this PR?**

/area networking
/kind enhancement

**What this PR does / why we need it**:
The `UseDNSRecords` feature gate in the apiserver, controllermanager and gardenlet has been promoted to `GA`.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```breaking operator
The `UseDNSRecords` feature gate in the `apiserver`, `controllermanager` and `gardenlet` has been promoted to `GA`. ```

### CODE REVIEW

File: pkg/gardenlet/features/features.go [hunk 1]
```diff
@@ -45,5 +45,5 @@ func RegisterFeatureGates() {
 
 // DisabledDNSProviderManagement returns true if `DisableDNSProviderManagement` is effective.
 func DisabledDNSProviderManagement() bool {
```
Comment (shafeeqes) [inline pkg/gardenlet/features/features.go]: Is this function really necessary now? I mean, can't we just replace its occurrences with `FeatureGate.Enabled(features.DisableDNSProviderManagement)` ?

**Review by stoyanr — DISMISSED** [review]
/lgtm

**Review by rfranzke — APPROVED** [review]
Nice!
/lgtm

**Pull request merged**